task4/VCS/Watcher.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace VCS
+{
+    /// <summary>
+    /// A class that monitors the file system
+    /// </summary>
+    public class Watcher
+    {
+        private DateTime lastChange;
+        private FileSystemWatcher watcher;
+
+        public Action<string> ChangeHandler;
+        public Commit Commit { get; private set; }
+        
+        public Watcher()
+        {
+            watcher = new FileSystemWatcher
+            {
+                Path = Storage.DirectoryToWatch,
+                Filter = "*.txt",
+                IncludeSubdirectories = true,
+                NotifyFilter = NotifyFilters.FileName |
+                NotifyFilters.LastWrite |
+                NotifyFilters.CreationTime,
+                EnableRaisingEvents = true,
+            };
```

Comment:
```suggestion
            watcher = new FileSystemWatcher
            {
                Path = Storage.DirectoryToWatch,
                Filter = "*.txt",
                IncludeSubdirectories = true,
                NotifyFilter = NotifyFilters.FileName
                    | NotifyFilters.LastWrite
                    | NotifyFilters.CreationTime,
                EnableRaisingEvents = true,
            };
```

(pipe operators at the end or start of the line in such case is more of a personal choice - depends on how you like to comment out / delete lines)